### PR TITLE
Expand use of and support for LTO and FDO builds

### PR DESCRIPTION
**Pull request by kcgen**
Adds LTO to the CI build for Linux, which bring it as close as possible to the planned formal release, which will additionally use FDO.

Adds some helper scripts to work with FDO files.

Improves the build notes for how to create and use FDO files.

### CODE REVIEW

**Comment by kcgen** [comment]
@dreamer , I think (?) this addresses your previous concerns. 
- Drops using FDO entirely in the workflow YAMLs (avoiding adding that complexity, and avoiding the dependence on the tests-repo)
- Adds extensive documentation about how to setup a system for AutoFDO capture so folks can run whatever training set they like; further disconnecting dependence on the test-repo as users can generate their own `.afdo` data. (I think this addresses your concern about  reproducibility and problems running the test-repo; which are still valid - but no longer tethered to this PR).
- FDO will now only be used via the build-script
- Replaces `parallel` with `xargs`
- Documentation now suggests installing `autofdo` via package manager, if available, and the scripts will make use of it as-is.

Most huge software packages are released with  FDO (plus LTO) - http://hubicka.blogspot.com/2019/05/gcc-9-link-time-and-inter-procedural.html (Firefox, LibreOffice, Google Chrome), and some distributions like Clear Linux are almost entirely built with LTO+FDO.

So If we do plan on using FDO for releases, we should get in the habit of using it and trusting it. After all, it just provides statistical data regarding branch probabilities versus the default guesses done by`-O2` or `-O3` (which are pretty good).

If we're worried that FDO data will actually harm the optimizations and cause our binary to go haywire (to the point where we don't even trust it for day-to-day builds) then we should be even more fearful of using it for actual releases. I know we don't believe that though - as there is no evidence to this fact; so we need to find a way to slowly start using it and earning our trust (if that's a concern).

File: scripts/build.md [hunk 1]
```diff
@@ -53,19 +53,19 @@ Use of both scripts is described below.
...
-Builds on Mac can be performed with Clang or GCC.
+Builds on macOS can be performed with Clang or GCC.
 
-If you only plan on only building with Clang, then follow the Brew
-installation steps. If you're interested in building with GCC, then either
-Brew or MacPorts will work. Both can be installed without conflicting with
-eachother.
+If you only plan on only building with Clang, then follow the Brew installation
+steps. If you're interested in building with GCC, then either Brew or MacPorts
+will work. Both can be installed without conflicting with eachother.
```
Comment (dreamer) [inline scripts/build.md]: "each other"

File: scripts/build.md [hunk 2]
```diff
@@ -101,59 +102,165 @@ agreed to:
...
+     ... for each `*.prof`.
+
+   - When you now have a pile of `*.afdo` files, merge them into a single
+     `curren.afdo` file with: `profile_merger -gcov_version=1
+     -output_file=current.afdo *.afdo`
+
+   For Clang, run:
+   
+   - `create_llvm_prof --binary=/path/to/fdo-trained/dosbox/binary
+     --profile=samples-1.prof --out=samples-1.profraw
```
Comment (dreamer) [inline scripts/build.md]: Missing tick in the end of line 255 (causes problems in following lines).

**Comment by kcgen** [comment]
Thanks for the review; will get these in!

**Comment by dreamer** [comment]
Squashed the two commits on this branch and merged as 78ae277d287c449a8ee5c0311dfe7c6ed29d8f8b.

Thanks! One step closer to the release :)

**Pull request closed without merging**